R_scripts/slope_calc_master_script.R

```diff
@@ -0,0 +1,299 @@
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # 
+# Function to select introns for slope calculation based on various criteria          #
+# Args:                                                                               #
+#   intron_info_file: Data frame containing information on introns                    #
+# Returns:                                                                            #
+#   Data frame filtered based on specific criteria, for downstream slope calculation. #
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+select_introns_for_slopecalc <- function(intron_info_file) 
+  {
+  intron_info_file <- intron_info_file %>%
+    filter(overlap_outside_reference != TRUE) %>%      # Exclude introns overlapping outside reference
+    filter(supported_by_samples == 1) %>%              # Include only introns supported by 100% of samples
+    filter(strand_measured != 0) %>%                   # Include only introns where the strand annotation is conclusive +/-
+    filter((start_match_reference == TRUE & end_match_reference == TRUE) |  # Fully match reference intro
```

Comment:
These 3 conditions can be probably rewritten as one using logical OR (start_match_reference == TRUE | end_match_reference == TRUE)